servers/src/main/java/tachyon/worker/lineage/LineageMasterClient.java

```diff
@@ -35,27 +35,28 @@
  * Since thrift clients are not thread safe, this class is a wrapper to provide thread safety, and
  * to provide retries.
  */
-public final class LineageMasterWorkerClient extends ClientBase {
+public final class LineageMasterClient extends ClientBase {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private LineageMasterService.Client mClient = null;
+  private LineageMasterWorkerService.Client mClient = null;
 
   /**
    * @param masterAddress the master address
    * @param tachyonConf tachyonConf
    */
-  public LineageMasterWorkerClient(InetSocketAddress masterAddress, TachyonConf tachyonConf) {
+  public LineageMasterClient(InetSocketAddress masterAddress, TachyonConf tachyonConf) {
     super(masterAddress, tachyonConf, "lineage-worker");
   }
 
   @Override
   protected String getServiceName() {
-    return Constants.LINEAGE_MASTER_SERVICE_NAME;
+    return Constants.LINEAGE_MASTER_WORKER_SERVICE_NAME;
   }
 
   @Override
-  protected void afterConnect() {
-    mClient = new LineageMasterService.Client(mProtocol);
+  protected void afterConnect() throws IOException {
+    mClient = new LineageMasterWorkerService.Client(mProtocol);
+    checkVersion(mClient, Constants.LINEAGE_MASTER_WORKER_SERVICE_VERSION);
```

Comment:
Currently, `checkVersion` is a concrete method in `ClientBase` so that the various clients can share its implementation. I am not sure what form of enforcement mechanism are you suggesting.